apps/wallet/src/background/account-sources/index.ts

```diff
@@ -72,6 +73,36 @@ async function createAccountSource({
 	}
 }
 
+async function deleteAccountSourceByType({
+	type,
+}: MethodPayload<'deleteAccountSourceByType'>['args']) {
+	switch (type) {
+		case 'mnemonic':
+			try {
+				const mnemonicAccountSources = await getAccountSources({ type: 'mnemonic' });
+				const sourceIds = mnemonicAccountSources.map((source) => source.id);
+				const accountPromises = sourceIds.map((id) => getAccountsBySourceID(id));
+				const allAccountsOfSources = await Promise.all(accountPromises);
+				const accountIds: string[] = [];
+				allAccountsOfSources.forEach((accForSource) => {
+					accForSource?.forEach((acc) => {
+						accountIds.push(acc.id);
+					});
+				});
+				const allImportedAccounts = await getAccountsByType('imported');
+				const allAccountIds = accountIds.concat(allImportedAccounts?.map((acc) => acc.id) || []);
+				await (await getDB()).accounts.bulkDelete(allAccountIds);
+				await (await getDB()).accountSources.bulkDelete(sourceIds);
```

Comment:
also seeing an issue when going through the recovery flow...

- small UI defect where clicking the Forgot Password button triggers / displays form validation -- we'll need to fix this up eventually so probably best to address that
- I'm seeing the screen stating that accounts not associated with the passphrase will be removed. However the account shown is associated with the passphrase as the first derived account: 

https://github.com/MystenLabs/sui/assets/122397493/0ba116f6-aa0e-4cd8-9124-9852b9da4058

Reply:
talked to pavlos about it, I think the wording probably needs to change here, but we should remove them all unless we start allowing separate passwords or something